### PR TITLE
[socat]: build socat with readline

### PR DESCRIPTION
**Pull request by lguohan**
support readline in bcmsh

Signed-off-by: Guohan Lu <gulv@microsoft.com>

**- What I did**
build socat with readline

**- How I did it**

**- How to verify it**
build image and use bcmsh

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
@zhenggen-xu

**Comment by lguohan** [comment]
address #1763

File: src/socat/Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+.SHELLFLAGS += -e
+
+MAIN_TARGET = socat_$(SOCAT_VERSION)_amd64.deb
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	# Remove any stale files
+	rm -rf ./socat-1.7.3.1
+
+	# Get source package
+	wget -NO socat_$(SOCAT_VERSION).dsc "https://sonicstorage.blob.core.windows.net/packages/debian/socat_1.7.3.1-2+deb9u1.dsc?sv=2015-04-05&sr=b&sig=Ph7aMqb%2F%2FE%2F8qwxMXoXb5oK1YPkfVt6PV8mBBv5Wi%2F4%3D&se=2155-07-05T11%3A42%3A29Z&sp=r"
```
Comment (zhenggen-xu) [inline src/socat/Makefile]: is it possible to use "dget" to simplify the steps to get and extract the source code?

Comment (lguohan) [reply src/socat/Makefile]: sometimes, debian update the package version causing our build break.

**Pull request merged**